src/styled_text.rs

```diff
@@ -31,24 +33,41 @@ impl StyledText {
         &self,
         insertion_point: usize,
         // prompt_style: &Style,
```

Comment:
Should we yank that, or do we want to transfer that through?

Reply:
I think that is a design decision that should be done "collectively"; does it make sense that the prompt style is forwarded like that? Do we want to take it out? Or should `StyledText` have access to it through other means?

Note that the Style is currently created inside the function because there was no way to obtain a compatible `Style` from the call sites, and the conversion between `nu_ansi_term::Style` and `crossterm::Style` wasn't easy to perform.

Reply:
Future separation of concern alternative would be to move the integration of the prompt strings out of the `render_around_insertion_point()` function and do that in the painter as the prompt is passed into that one already. Writing a translation between the `nu-ansi-term` and `crossterm` styles was on my list of coding tasks to mindlessly do with rust analyzer :stuck_out_tongue_winking_eye:

Reply:
If you're okay with that, then, I would remove the "comment-parameter" and leave the conversion task to you! ;P

Reply:
The conversion turns out to be not completely 1 to 1 as nu-ansi-term seems to use the full xterm256 8-bit color palette to define its main colors while crossterm seems to rely on the 4-bit basic set (that might be more cross plattform) for the named colors. But we might have to check in with the nu old guard what their experience was so far.